### DISCUSSION TITLE
Record History

### DISCUSSION THREAD

**Discussion by djbbiz** [cortexrd/Knack-Toolkit-Library#341] [Q&A]
I am using the record history and find it very useful albeit it needs to be controlled due to the volume of potential edits that can happen to a table of records. One of the things I would like to do, is limit the exact fields that go into the Before and After inside the Changes field. I have 4 or 5 fields that are doing calculations and changing but are meaningless to the history. The only thing I want to see is a specific final field I designate for the before and after. I don't see a way to do that but can that be added as a parameter to the _arh?

**Comment by cortexrd** [comment]
Hi Dean,

So, I assume you saw the exclude fields parameter, but it's not what you want?

I'm referring to this:

_arh=[context, string], [include, fields], **[exclude, fields]**, [expiry, delay]

What I understand is that you don't want to see those fields+values in the recorded output short text field called Changes - right?
In this case, if it was supported to remove some fields from the output, it would not use less records.  Just less string space.

Just to be sure I understand, can you send us a screen grab of what you have vs what you'd like to have?

Norm

**Reply by djbbiz** [reply]
You are correct, using less records was just a comment that I noted is an issue when you have high volume of traffic and many changes as we do in our task system for example. But that being said, when you have high volume, all the fields you don't need to see in the Change field can create a lot of noise in your view making you hunt for the data you want to see. So if Field_1 = a and field_2 = b and field_3 = field_1 + field_2 for example, I may only want to see what field_3 changes were and not the variables that were changed that went into it. Having the option to show them when I want by adding the fields I want to show in a parameter allows me to see details when I am debugging or just show a result field as a view for a user.  So the parameter could be show all changes when it isn't set as the default view, but only shows the fields requested when the parameter is set {field_3, field_4]

**Reply by cortexrd** [reply]
I see what you mean, and it makes sense.  Actually, it should have been like that since the beginning.  It's just a use case I didn't think of.
I'll find a fix soon,
Norm
